sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/WebSocketsProxyConnectionHandler.java

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.implementation.handler;
+
+import com.azure.core.amqp.implementation.AmqpErrorCode;
+import com.azure.core.amqp.models.ProxyConfiguration;
+import com.azure.core.implementation.util.ImplUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.microsoft.azure.proton.transport.proxy.ProxyAuthenticationType;
+import com.microsoft.azure.proton.transport.proxy.ProxyHandler;
+import com.microsoft.azure.proton.transport.proxy.impl.ProxyHandlerImpl;
+import com.microsoft.azure.proton.transport.proxy.impl.ProxyImpl;
+import org.apache.qpid.proton.amqp.transport.ConnectionError;
+import org.apache.qpid.proton.amqp.transport.ErrorCondition;
+import org.apache.qpid.proton.engine.Connection;
+import org.apache.qpid.proton.engine.Event;
+import org.apache.qpid.proton.engine.Transport;
+import org.apache.qpid.proton.engine.impl.TransportInternal;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Creates an AMQP connection using web sockets (port 443) and connects through a proxy.
+ */
+public class WebSocketsProxyConnectionHandler extends WebSocketsConnectionHandler {
+    private static final String HTTPS_URI_FORMAT = "https://%s:%s";
+    private static final String PROXY_SELECTOR_HAS_BEEN_MODIFIED = "ProxySelector has been modified.";
+
+    private final ClientLogger logger = new ClientLogger(WebSocketsProxyConnectionHandler.class);
+    private final String amqpHostname;
+    private final ProxyConfiguration proxyConfiguration;
+
+    /**
+     * Creates a handler that handles proton-j's connection through a proxy using web sockets.
+     *
+     * @param connectionId Identifier for this connection.
+     * @param amqpHostname Hostname of the AMQP message broker. The hostname of the proxy is exposed in {@link
+     *     #getHostname()}.
+     * @throws NullPointerException if {@code amqpHostname} or {@code proxyConfiguration} is null.
+     */
+    public WebSocketsProxyConnectionHandler(String connectionId, String amqpHostname,
+                                            ProxyConfiguration proxyConfiguration) {
+        super(connectionId, amqpHostname);
+        this.amqpHostname = Objects.requireNonNull(amqpHostname, "'amqpHostname' cannot be null.");
+        this.proxyConfiguration = Objects.requireNonNull(proxyConfiguration, "'proxyConfiguration' is required.");
```

Comment:
Use the same error message to be consistent. `'proxyConfiguration' cannot be null.`